python/core/auto_generated/providers/arcgis/qgsarcgisportalutils.sip.in

```diff
@@ -32,7 +32,7 @@ Utility functions for working with ArcGIS REST services.
       ImageService,
     };
 
-    static QVariantMap retrieveUserInfo( const QString &communityUrl, const QString &user, const QString &authcfg, QString &errorTitle /Out/, QString &errorText /Out/, const QMap< QString, QString > &requestHeaders = QMap< QString, QString >(), QgsFeedback *feedback = 0 );
+    static QVariantMap retrieveUserInfo( const QString &communityUrl, const QString &user, const QString &authcfg, QString &errorTitle /Out/, QString &errorText /Out/, const QgsHttpHeaders &requestHeaders = QgsHttpHeaders(), QgsFeedback *feedback = 0 );
```

Comment:
Can you confirm that sip is correctly automatically convert string maps to the new class here? Otherwise it's an API break.,.

Reply:
I  will check this part.

Reply:
@nyalldawson this does not work. I have to encapsulate the map in the class 
```python
        headers = QgsHttpHeaders( {'referer' : 'http://google.com'} )
        res = QgsArcGisPortalUtils.retrieveUserInfo('http://' + endpoint, 'some_user', '', headers )
```

At the beginning this new class inherited QMap but SIP is not able to handle it!
Is there a solution to have SIP working with QMap inherited class?

Reply:
> Is there a solution to have SIP working with QMap inherited class?

I honestly have no idea. I think it'll be simplest to just leave the method with the existing signature, and add a new method which uses the QgsHttpHeaders class

Reply:
Ok... so bad inheritance is a bit buggy with SIP :disappointed: 

 I add wrap functions and I test in the python test.